### PR TITLE
Fix progress bar compatibility with generators

### PR DESCRIPTION
**Pull request by brody715**
## Description

- Fix to make progress bar compatiable with generators

image

Closes: #3841

There's no attribute `__len__` in generators, and even when users explicitly set the `total` argument, it defaults to None, preventing proper functionality with generators.

So we will use `total` as `length` if the object has no attribute `__len__`. 

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: gradio/helpers.py [hunk 1]
```diff
@@ -764,7 +764,7 @@ def tqdm(
                 self.iterables.append(new_iterable)
                 callback(self.iterables)
                 return self
-            length = len(iterable) if hasattr(iterable, "__len__") else None  # type: ignore
```
Comment (freddyaboulton) [inline gradio/helpers.py:767]: Looks good to me @brody715 - can you please format the code with ruff (`bash scripts/format_backend.sh`)? We can merge it in then

Comment (brody715) [reply gradio/helpers.py:767]: sure! But I cannot access to my PC now. I will do it later.

Comment (brody715) [reply gradio/helpers.py:767]: @freddyaboulton Done

**Comment by freddyaboulton** [comment]
Thank you for the contribution @brody715

**Pull request merged**